### PR TITLE
CEDS-1667 Support for CHIEF errors

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class ErrorSpec extends UnitSpec {
+
+  "Error apply method" should {
+
+    "return an Error when the structure is correct" in {
+
+      val inputList = List("code", "description")
+      val expectedError = Error("code", "description")
+
+      Error.apply(inputList) shouldBe expectedError
```
Comment (ed0906) [inline test/unit/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorSpec.scala]: `Error(inputList)`?

Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorSpec.scala]: Fixed

File: app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+package uk.gov.hmrc.exports.movements.models.notifications.parsers
+
+import java.util.regex.Pattern
+
+import com.github.tototoshi.csv._
+import play.api.Logger
+
+import scala.io.Source
+
+case class Error(code: String, description: String)
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala]: We should not use descriptionthis way. It is better to use message keys, as we will eventually need to implement Welsh.

File: app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  private val chiefErrors: List[Error] = {
+    val source = Source.fromURL(getClass.getClassLoader.getResource("chief_errors.csv"), "UTF-8")
+
+    readErrorsFromFile(source)
+  }
+
+  private def retrieveChiefErrorCode(errorMessage: String): Option[String] =
+    errorMessage.split(" ").filter(chiefErrorPattern.matcher(_).matches).headOption
+
+  def validateErrors(errors: Seq[String]): Seq[String] = {
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala]: As per 'Clean code' guidance, it is better to have public methods first and private methods after.

File: app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/notifications/parsers/ErrorParser.scala]: So we're reading descriptions from the file in that case it's doesn't make any sense to rewrite it to messages.en. I'll add sticker about it to the board, and we can discuss it

**Pull request merged**